### PR TITLE
Upgrade to go 1.18

### PR DESCRIPTION
**Pull request by cammellos**
Upgrade to golang 1.18

### CODE REVIEW

**Comment by vitvly** [comment]
This does not seem related to this branch specifically, but when running `make vendor` i get this:
```
➜  status-go git:(feature/go-upgrade-1.18) make vendor
go mod tidy
go mod vendor
modvendor -copy="**/*.c **/*.h" -v
Error! "/Users/vvlasov/c/status-go-root/pkg/mod/explicit;@go" module path does not exist, check $GOPATH/pkg/mod
make: *** [vendor] Error 1
```

File: go.mod [hunk 1]
```diff
@@ -76,8 +69,209 @@ require (
...
 	golang.org/x/crypto v0.0.0-20220131195533-30dcbda58838
 	golang.org/x/image v0.0.0-20210220032944-ac19c3e999fb
-	gopkg.in/go-playground/assert.v1 v1.2.1 // indirect
+	google.golang.org/protobuf v1.27.1
 	gopkg.in/go-playground/validator.v9 v9.31.0
 	gopkg.in/natefinch/lumberjack.v2 v2.0.0
 	olympos.io/encoding/edn v0.0.0-20201019073823-d3554ca0b0a3
 )
+
+require (
```
Comment (jakubgs) [inline go.mod:78]: Why is there a second `require` section?

Comment (jakubgs) [reply go.mod:78]: Apparently:

>After a suitable module has been found, the go command will add a new [requirement](https://go.dev/ref/mod#go-mod-file-require) with the new module’s path and version to the main module’s go.mod file. This ensures that when the same package is loaded in the future, the same module will be used at the same version. __If the resolved package is not imported by a package in the main module, the new requirement will have an `// indirect` comment.__

https://go.dev/ref/mod#resolve-pkg-mod

Comment (jakubgs) [reply go.mod:78]: >The go command automatically adds `// indirect` comments for some requirements. An `// indirect` comment indicates that no package from the required module is directly imported by any package in the [main module](https://go.dev/ref/mod#glos-main-module).

https://go.dev/ref/mod#go-mod-file-require

Comment (cammellos) [reply go.mod:78]: mmh, good point, I think something went funny during the rebase

**Comment by cammellos** [comment]
> This does not seem related to this branch specifically, but when running `make vendor` i get this:
> 
> ```
> ➜  status-go git:(feature/go-upgrade-1.18) make vendor
> go mod tidy
> go mod vendor
> modvendor -copy="**/*.c **/*.h" -v
> Error! "/Users/vvlasov/c/status-go-root/pkg/mod/explicit;@go" module path does not exist, check $GOPATH/pkg/mod
> make: *** [vendor] Error 1
> ```

could you try to upgrade mod-vendor?
`make install-modvendor`

**Review by jakubgs — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (jakubgs) [reply go.mod:78]: Seems to work fine, so whatever.

**Comment by vitvly** [comment]
@cammellos beautiful, thanks, problem gone :)

**Comment by Samyoul** [comment]
Ah, ok I'm happy to see this split into its own PR

**Pull request merged**